### PR TITLE
Add SquimSubjective Model

### PR DESCRIPTION
**Pull request by nateanl**
Add model architecture and factory functions for `SquimSubjective` which predicts subjective evaluation metric scores (e.g. MOS) for speech enhancement task.

### CODE REVIEW

**Comment by nateanl** [comment]
cc @JupiterEthan

File: torchaudio/prototype/models/squim/subjective.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        x = self.att_pool_layer(x)
+        x = nn.functional.softmax(x, dim=1)
+        B = torch.linspace(0, 4, steps=self.att_dim, device=x.device)
+        x = (x * B).sum(dim=1)
+        return x
+
+
+class SquimSubjective(nn.Module):
+    """Speech Quality and Intelligibility Measures (SQUIM) model that predicts **subjective** metric scores
+    for speech enhancement (e.g., MOS).
```
Comment (mthrok) [inline torchaudio/prototype/models/squim/subjective.py:71]: Is there a reference we should add??

**Review by mthrok — APPROVED** [review]

File: torchaudio/prototype/models/squim/subjective.py (hunk 1 above)
Comment (nateanl) [reply torchaudio/prototype/models/squim/subjective.py:71]: There is. I forgot to add it here. Will make a quick fix.
